spectra-generator/src/main/java/se/de/hu_berlin/informatik/gen/spectra/predicates/pdg/CodeLocation.java

```diff
@@ -0,0 +1,39 @@
+package se.de.hu_berlin.informatik.gen.spectra.predicates.pdg;
+
+import soot.SootMethod;
+import soot.Unit;
+
+public class CodeLocation {
+
+    public Unit unit;
+    public String className;
+    public String methodName;
+    public SootMethod method;
+    public SootConnector sootConnector;
+
+    public CodeLocation(Unit unit, String className, SootMethod method, SootConnector sootConnector)
+    {
+        this.unit = unit;
+        this.className = className;
+        this.methodName = method.getName();
+        this.method = method;
+        this.sootConnector = sootConnector;
+    }
+
+    public String getLocationString()
+    {
+        return this.className + ":" + this.unit.getJavaSourceStartLineNumber();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj instanceof CodeLocation) {
+            CodeLocation other = (CodeLocation) obj;
+            if (other.getLocationString().equals(this.getLocationString()))
```

Comment:
if you compare items regularly, this line might trigger the concatenation of Strings performed in getLocationString a lot of times. For performance reasons, it would make more sense to either store htis concatenated String as a class variable or to maybe compare the class names and line numbers in separate operations.

Of course, I don't really know how much is being optimized automatically by the compiler...